### PR TITLE
DevTools: Debug C/C++ Wasm

### PR DESCRIPTION
**Pull request by sofiayem**
Documented Wasm debugging in DevTools.

@tomayac, @pfaffe Would you kindly fact-check and provide feedback on my doc?

### CODE REVIEW

File: site/en/docs/devtools/wasm/index.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+---
+
+[WebAssembly](https://developer.mozilla.org/docs/WebAssembly) provides a way to run, for example, C/C++ code on the web at near native speed and alongside JavaScript. This document shows how to set up and use Chrome DevTools to better debug such applications.
+
+{% YouTube id='VBMHswhun-s', startTime='170' %}
+
+Once you set up DevTools, you can:
+
+- Inspect your original code in **Sources** > **Editor**.
+- Pause execution with line-of-code [breakpoints](/docs/devtools/javascript/breakpoints/#loc) and step through your original source code, not the `.wasm` binary file.
```
Comment (tomayac) [inline site/en/docs/devtools/wasm/index.md:21]:
```suggestion
- Pause execution with line-of-code [breakpoints](/docs/devtools/javascript/breakpoints/#loc) and step through your original C/C++ source code, not the compiled `.wasm` binary file.
```

File: site/en/docs/devtools/wasm/index.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+{% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/ieeESEUXLsEsP7jl7evq.png", alt="A nested variable and a complex object expanded in the Console.", width="800", height="668" %}
+
+- In **Sources** > **Scope**, click the {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/jgJz0iMIw4XhDqhdwYP2.svg", alt="Memory.", width="20", height="20" %} icon to open the **Memory Inspector** and inspect the raw bytes of object memory. For more information, see [WebAssembly memory inspection](/docs/devtools/memory-inspector/#wasm).
+
+{% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/mD5y0xEJJ7kCfclIGwWS.png", alt="Inspecting the memory of a variable.", width="800", height="633" %}
+
+## Profile performance {: #perf }
+
+With DevTools set up and open, the code Chrome runs isn't optimized. It's tiered down to give you better debugging experience.
```
Comment (tomayac) [inline site/en/docs/devtools/wasm/index.md:81]:
```suggestion
With DevTools set up and open, the code Chrome runs isn't optimized. It's [tiered down](https://v8.dev/docs/wasm-compilation-pipeline#debugging) to give you better debugging experience.
```

File: site/en/docs/devtools/wasm/index.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    ```bash
+    emcc -g source.cc -o app.html
+    ```
+1. Install the [C/C++ DevTools Support (DWARF) Chrome extension](https://goo.gle/wasm-debugging-extension).
+
+## Debug {: #debug }
+
+With DevTools set up, debug your code:
+
+1. [Open DevTools](/docs/devtools/open/) to inspect your website. For this tutorial, you can try it on this [demo page](https://emscripten-dbg-stories.netlify.app/mandelbrot.html).
```
Comment (tomayac) [inline site/en/docs/devtools/wasm/index.md:45]:
```suggestion
1. [Open DevTools](/docs/devtools/open/) to inspect your website. For this tutorial, you can try it on this [demo page](https://emscripten-dbg-stories.netlify.app/mandelbrot.html), which was compiled with the required `-g` flag.
```

**Review by tomayac — APPROVED** [review]
LGTM, this is super detailed. Left a couple of suggestions.

**Review by rachelandrew — APPROVED** [review]

File: site/en/docs/devtools/wasm/index.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+With DevTools set up and open, the code Chrome runs isn't optimized. It's [tiered down](https://v8.dev/docs/wasm-compilation-pipeline#debugging) to give you better debugging experience.
+
+In this case, you can't rely on `console.time()` and `performance.now()`  in your code to profile performance. Instead, use the [**Performance**](/docs/devtools/performance/reference/#record) panel to profile.
+
+Alternatively, you can run your profiling code without opening DevTools, then open it to inspect the messages in the **Console**.
+
+## Separate the debug information {: #separate-debug }
+
+To speed up compilation and loading times but still have a better debugging experience, you can split out the debug information into a separate `.wasm` file. You can either keep this file locally or host it on a separate server.
```
Comment (pfaffe) [inline site/en/docs/devtools/wasm/index.md:89]: -gseparate-dwarf arguably only affects loading time, not compilation. Would it make sense to link to [Debugging WebAssembly Faster] here, or is it enough that that's referenced at the bottom?

**Review by pfaffe — APPROVED** [review]
Looks great, thank you!

**Pull request merged**